### PR TITLE
getHeaders()

### PR DESCRIPTION
**Pull request by AdamGaskins**
PR for #51.

Love this package, and thanks for all the value you've added to the community (Laravel Beyond Crud has had a huge impact on our company 🙌🏻 ).

### CODE REVIEW

**Review by freekmurze — CHANGES_REQUESTED** [review]
Let's make the code a bit simpler!

**Comment by AdamGaskins** [comment]
My understanding is the spreadsheet needs to be opened and the headers processed *after* any of the fluent methods are called (`useDelimiter()`, `trimHeaderRow()` etc.) but *before* either of `getRows()` or `getHeaders()` is called (e.g. we have a page on our site that opens the spreadsheet solely to read the headers for mapping purposes). Extracting this processing to its own method seemed to make the most sense to me, and also avoided `getHeaders()` needing to rewind the iterator. Let me know your thoughts!

**Comment by freekmurze** [comment]
I think the clearest things to do, would be to have `getHeaders` to open the file and have it's own iterator independent of the `getRows` function. This would you won't need to keep a boolean tracking whether things have been processed or not.

I really don't mind that the file is opened twice, the performance impact will probably be negligible. Typing this message likely takes more time 🙂

File: src/SimpleExcelReader.php [hunk 1]
```diff
@@ -154,6 +161,31 @@ public function getRows(): LazyCollection
         });
     }
 
+    public function getHeaders(): ?array
+    {
+        if ($this->processHeader && ! $this->headers) {
```
Comment (freekmurze) [inline src/SimpleExcelReader.php:166]: There's a big piece of indented code here. I can be simplified by first checking if headers is set to something. If it is, we can already return the headers.

**Review by freekmurze — CHANGES_REQUESTED** [review]

**Comment by freekmurze** [comment]
Thank you!

**Comment by avdb** [comment]
Is there a reason why $reader is created in the getHeaders()?
I think you cannot set the delimiter for csv files how it's now implemented but maybe I'm missing something!

**Pull request merged**